pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -117,12 +119,39 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
 		{Event: framework.ClusterEvent{Resource: framework.CSINode, ActionType: framework.Add | framework.Update}},
 		// When CSIStorageCapacity is enabled, pods may become schedulable
 		// on CSI driver & storage capacity changes.
-		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterCSIDriverChange},
 		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}},
 	}
 	return events
 }
 
+func (pl *VolumeBinding) isSchedulableAfterCSIDriverChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	_, modifiedCSIDriver, err := util.As[*storagev1.CSIDriver](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"CSIDriver", klog.KObj(modifiedCSIDriver),
+	)
+
+	for _, vol := range pod.Spec.Volumes {
+		if vol.CSI == nil || vol.CSI.Driver != modifiedCSIDriver.Name {
+			continue
+		}
+		driverSpec := modifiedCSIDriver.Spec
+		if driverSpec.StorageCapacity != nil && *driverSpec.StorageCapacity {
```

Comment:
Is it the opposite?

```suggestion
		if driverSpec.StorageCapacity == nil || !*driverSpec.StorageCapacity {
```

---

When `driver.Spec.StorageCapacity == nil || !*driver.Spec.StorageCapacity`, we return true on `hasEnoughCapacity`.
https://github.com/kubernetes/kubernetes/blob/8565e375251450a291f0af3c6195c7a5bf890292/pkg/scheduler/framework/plugins/volumebinding/binder.go#L1022-L1024

Meaning, if driver's `StorageCapacity` is modified to match `driver.Spec.StorageCapacity == nil || !*driver.Spec.StorageCapacity`, the modification could make Pod schedulable, right?

Reply:
@sanposhiho 
I apologize for my misunderstanding!
I have corrected the issues.[e225283](https://github.com/kubernetes/kubernetes/pull/125171/commits/e225283f9d20284d9f404abfa412ebbc37f44e0c)